### PR TITLE
chore: update Dockerfile

### PR DESCRIPTION
**Pull request by themr0c**
## What does this pull request change?

* Update Dockerfile.
* New version for all tools: antora, newdoc, shellcheck, vale ...

## What issues does this pull request fix or reference?

* Quay reports container image has security issues.
* Container image doesn't build.

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
Would be better to use UBI images but +1

**Review by benoitf — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -9,36 +9,37 @@
...
 #
 
-FROM rust:1.49.0-alpine3.12 as newdoc-builder
-RUN cargo install newdoc
+FROM docker.io/library/rust:1.51-alpine3.13 as newdoc-builder
+RUN apk add --no-cache --update  musl-dev \
+    && cargo install newdoc
 
-FROM golang:1.15.6-alpine3.12 as htmltest-builder
+FROM docker.io/library/golang:1.15-alpine3.13 as htmltest-builder
```
Comment (benoitf) [inline Dockerfile:16]: 1.15 is a rolling tag so usually we stick to a tag that is not changing

Comment (themr0c) [reply Dockerfile:16]: You mean to make sure we can rebuild exactly the same? 
I see this here as a feature to enable to build newer images including security fixes without changing the Dockerfile in master.

Comment (benoitf) [reply Dockerfile:16]: yes, to have something that is reproduced all the time

about security fixes updates, probably that dependabot can provide such enhancements ?

**Pull request merged**